pkg/apitype/policy.go

```diff
@@ -64,12 +68,31 @@ type Policy struct {
 	DisplayName string `json:"displayName"`
 
 	// Description is used to provide more context about the purpose of the policy.
-	Description      string           `json:"description"`
+	Description string `json:"description"`
+
+	// This enforcement level will be deprecated in favor of enforcement
+	// level specified by the default configuration.
 	EnforcementLevel EnforcementLevel `json:"enforcementLevel"`
 
 	// Message is the message that will be displayed to end users when they violate
 	// this policy.
 	Message string `json:"message"`
+
+	// DefaultConfiguration is used to specify the default configuration
+	// for the Policy.
+	DefaultConfiguration PolicyConfiguration `json:"defaultConfiguration"`
```

Comment:
I would think we'd want a different type for the policy pack's metadata.

For `https://blog.golang.org/godoc-documenting-go-code`, returning a `PolicyConfiguration` object makes sense. As it is the configuration to be applied to that policy when executed at runtime.

But when describing the `apitype.Policy` as part of a pack and push operation, I would think we need a different API shape. For example, the `DefaultConfiguration.Disabled` field doesn't make any sense. (Pack and push a policy that is disabled by default?!?)

Also, by calling this "default configuration" it isn't clear whether the Service would pass the same config map to the policy at runtime, even if the user hasn't customized the policy pack.

What do you think about the following shape instead, which just details the metadata for configuration values that the policy expects:

```
// ConfigurationMetadata describes the configuration values this policy expects.
ConfigurationMetadata PolicyConfigurationMetadata `json:"configurationMetadata"`

...
// PolicyConfigurationValue describes a single configuration value a policy uses.
type PolicyConfigurationValue struct {
    // Description of what the configuration value does.
    // "comma-separated ARN list of elastic IPs"
    // "AWS region requirement"
    // "Maximum number of EC2 machines"
    Description string


    // Key is the name configuration value. For example:
    // "supportedArns"
   // "targetRegion"
   // "maxVMs"
    Key string

    // If we want to get fancier later, we can add fields like:
    // Type, Required, Optional, DefaultValue, etc.
    //
    // But for now, just having a list of keys (to show in the service UI to be editable)
    // and a description of what those keys are used for seems sufficient.
}

// PolicyConfigurationMetadata describes the configuration data a policy supports at runtime.
type PolicyConfigurationMetadata {
    Values [] PolicyConfigurationValue `json:"values"`
}
```

Reply:
Part of the proposal was to require a default value for all Policy Packs, so they can always run. What are your thoughts on this?

Reply:
I went with`PolicyConfigurationVariable` >`PolicyConfigurationValue`